tests/unit/s2n_tls13_zero_length_payload_test.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_config.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls.h"
+#include "tls/s2n_tls13.h"
+#include "tls/s2n_record.h"
+#include "tls/s2n_handshake_io.c"
+
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <s2n.h>
+#include <fcntl.h>
+#include <sys/wait.h>
+
+/* In TLS 1.3, encrypted handshake records would appear to be of record type TLS_APPLICATION_DATA.
+*  The actual record content type is found after
+*/
+const char tls13_zero_length_application_record_hex[] = "170303000117";
+const char tls13_zero_length_handshake_record_hex[] = "1603030000";
+const char tls13_zero_length_alert_record_hex[] = "1503030000";
+
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    EXPECT_SUCCESS(s2n_enable_tls13());
+
+    /* Test 0 length application data record handled gracefully in client and server mode */
```

Comment:
Since the client and server aren't interacting here, it might be clearer if you split this into two separate tests. Same for the others.

Reply:
I personally found it more clear to see them side by side as they are almost identical. You can see that they are being executed in the same manner easily. However, if you feel strongly for the other way, it's a simple change.